### ISSUE TITLE
[Bug] be使用ipv6地址，进行ssb压测时，全部挂掉

### ISSUE THREAD

**Issue by xleoken** [apache/doris#40391]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1. version, 测试过2.1.5 3.0.1 版本

2. hosts 配置信息
```
3409:8d20:1833:da29::19 DORIS-CMCC-001
3409:8d20:1833:da29::11 DORIS-CMCC-002
```

3. 添加 BE，也有配置 enable_fqdn_mode = true
```
ALTER SYSTEM ADD BACKEND "DORIS-CMCC-001:9050";
ALTER SYSTEM ADD BACKEND "DORIS-CMCC-002:9050";
```

4. 执行 ssb 压测， https://doris.apache.org/docs/benchmark/ssb

在执行过程中，后端BE会挂掉

5. 日志

[fe-001.log.tar.gz](https://github.com/user-attachments/files/16873405/fe-001.log.tar.gz)

[be-001.log.tar.gz](https://github.com/user-attachments/files/16873414/be-001.log.tar.gz)

[be-002.log.tar.gz](https://github.com/user-attachments/files/16873415/be-002.log.tar.gz)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by yuanyuan8983** [comment]
Hello, if it is convenient to add my wechat, we will have a look together, my wechat signal is: Liyy_222_8993

**Comment by yuanyuan8983** [comment]
Explain the above questions:
brpc does not support ipv6 hostname resolution, resulting in connection initialization failure, nullptr is returned.
Calling get_client without checking nullptr results in coredump.
This PR fixes the nullptr check : https://github.com/apache/doris/pull/40074

**Comment by xleoken** [comment]
> Explain the above questions: brpc does not support ipv6 hostname resolution, resulting in connection initialization failure, nullptr is returned. Calling get_client without checking nullptr results in coredump. This PR fixes the nullptr check : #40074

好的，等新版发布之后，我验证一下
